### PR TITLE
Rocket Chat Integration Branch

### PR DESCRIPTION
**Pull request by patricksimonian**
## Summary
This pr serves as the integration for the rocket chat search feature

### CODE REVIEW

File: app-web/src/utils/hooks.js [hunk 1]
```diff
@@ -92,3 +111,55 @@ export const useImplicitAuth = () => {
...
+  useEffect(() => {
+    setLoading(loading);
+    if (!authenticated || !queryString) {
+      setResults([]);
+    } else if (!_loading) {
+      setResults(data.search);
+    }
+
+    return () => {
+      setResults([]);
```
Comment (ShellyXueHan) [inline app-web/src/utils/hooks.js]: so it doesn't matter is it's loaded, will it eventually return the setResults([]); ?

File: app-web/src/pages/index.js [hunk 2]
```diff
@@ -192,28 +202,53 @@ export const Index = ({
...
-        {siphonResources}
-      </Aux>
-    );
+    const { rocketchat } = searchSourceResults;
+    if (!!rocketchat.results) {
+      content = (
+        <Aux>
+          {getTopicPreviews(topics, windowHasQuery && !queryIsEmpty)}
+          {siphonResources}
+          {rocketchat.results.length > 0 && (
```
[the remark this answers is not in the document]
Comment (DerekSgov) [reply app-web/src/pages/index.js]: very good call

File: app-web/src/utils/hooks.js [hunk 3]
```diff
@@ -13,14 +13,33 @@ Created by Patrick Simonian
...
 import { createIam } from '../auth';
 import { isLocalHost } from './helpers';
 import { SEARCH_FIELD_NAMES, SEARCH_FIELD_MAPPING } from '../constants/search';
+import isEmpty from 'lodash/isEmpty';
+import AuthContext from '../AuthContext';
+import { useQuery } from '@apollo/react-hooks';
 import { ROCKET_GATE_QUERY } from '../constants/runtimeGraphqlQueries';
+
+function deepCompareEquals(a, b) {
+  return isEqual(a, b);
```
[the remark this answers is not in the document]
Comment (DerekSgov) [reply app-web/src/utils/hooks.js]: what exactly do you mean?

[the remark this answers is not in the document]
Comment (DerekSgov) [reply app-web/src/utils/hooks.js]: like why not just do !isEqual(value, ref.current)?

**Pull request merged**